Source/Scene/Expression.js

```diff
@@ -25,6 +27,18 @@ define([
 
     var scratchColor = new Color();
 
+    var scratchColorIndex = 0;
```

Comment:
Good catch.

For better cohesion, can you put this into a small static class (still in this file), e.g., with an `initialize`/`reset`/whatever function that sets this to zero, and then the `getScratchColor` function?